### ISSUE TITLE
KeyError: 'location'

### ISSUE THREAD

**Issue by Awmusic12635** [isFakeAccount/psnawp#6]
Seem to be getting this weird error about a location header missing when setting up the client. 

```
03:59:46 Traceback (most recent call last):
File "/app/.heroku/python/lib/python3.8/site-packages/rq/worker.py", line 936, in perform_job
rv = job.perform()
File "/app/.heroku/python/lib/python3.8/site-packages/rq/job.py", line 684, in perform
self._result = self._execute()
File "/app/.heroku/python/lib/python3.8/site-packages/rq/job.py", line 690, in _execute
return self.func(*self.args, **self.kwargs)
File "/app/mvp/jobs/gameaccount.py", line 225, in update_psn_account_info
client = psnawp.PSNAWP(os.getenv('PSNKEY'))
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/psnawp.py", line 13, in __init__
self.authenticator = authenticator.Authenticator(npsso_token=npsso)
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/authenticator.py", line 36, in __init__
self.authenticate()
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/authenticator.py", line 158, in authenticate
location_url = response.headers['location']
File "/app/.heroku/python/lib/python3.8/site-packages/requests/structures.py", line 54, in __getitem__
return self._store[key.lower()][1]
KeyError: 'location'
Traceback (most recent call last):
File "/app/.heroku/python/lib/python3.8/site-packages/rq/worker.py", line 936, in perform_job
rv = job.perform()
File "/app/.heroku/python/lib/python3.8/site-packages/rq/job.py", line 684, in perform
self._result = self._execute()
File "/app/.heroku/python/lib/python3.8/site-packages/rq/job.py", line 690, in _execute
return self.func(*self.args, **self.kwargs)
File "/app/mvp/jobs/gameaccount.py", line 225, in update_psn_account_info
client = psnawp.PSNAWP(os.getenv('PSNKEY'))
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/psnawp.py", line 13, in __init__
self.authenticator = authenticator.Authenticator(npsso_token=npsso)
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/authenticator.py", line 36, in __init__
self.authenticate()
File "/app/.heroku/python/lib/python3.8/site-packages/psnawp_api/authenticator.py", line 158, in authenticate
location_url = response.headers['location']
File "/app/.heroku/python/lib/python3.8/site-packages/requests/structures.py", line 54, in __getitem__
return self._store[key.lower()][1]
KeyError: 'location'
```

I don't "think" my token is expired because I tried setting some logging around that but i'm hoping someone else might have another idea

**Comment by isFakeAccount** [comment]
Actually doing logging around can make your token expire because sony will assign you new session cookie.

**Comment by Awmusic12635** [comment]
The token isn't expired, I have confirmed that.

**Comment by isFakeAccount** [comment]
@Awmusic12635 maybe it has to do eith heroko. Can you check what response header you get at line

**Comment by Awmusic12635** [comment]
I was able to fork it and add some logging. It seems it sometimes works but other times it doesn't. 

```
<HTML><HEAD>
 <TITLE>Access Denied</TITLE>
 </HEAD><BODY>
 <H1>Access Denied</H1>

 You don't have permission to access "http&#58;&#47;&#47;ca&#46;account&#46;sony&#46;com&#47;api&#47;authz&#47;v3&#47;oauth&#47;authorize&#63;" on this server.<P>
Reference&#32;&#35;18&#46;142be8ac&#46;1618957948&#46;2c898f64
</BODY>
</HTML>

{'Server': 'AkamaiGHost', 'Mime-Version': '1.0', 'Content-Type': 'text/html', 'Content-Length': '322', 'Expires': 'Tue, 20 Apr 2021 22:32:28 GMT', 'X-Reference-Error': '18.142be8ac.1618957948.2c898f64', 'Date': 'Tue, 20 Apr 2021 22:32:28 GMT', 'Connection': 'close'}
```

**Comment by isFakeAccount** [comment]
maybe the Akamai bot detector does not like the Heroku and basically IP banning you. You can read more here

https://www.akamai.com/us/en/products/security/bot-manager.jsp

**Comment by isFakeAccount** [comment]
@Awmusic12635 Did your solution work out? Any updates?

**Comment by isFakeAccount** [comment]
To test this particular bug I hosted one of my bots that uses this API on heroku. So far I have not encountered this particular bug. I am closing this issue since it is not reproducible and on top of that, OP is not responding.

**Comment by isFakeAccount** [comment]
This issue is now officially fixed in the latest release 2021.20.6
